cc1120_arduino/cc1120_spi.c

```diff
@@ -8,32 +8,35 @@
  * @param addr - The address of the first register to read.
  * @param data - The array to store the read data, or a pointer to a single uint8_t if len=1.
  * @param len - The number of registers to read.
- * @return true - If the read was successful.
- * @return false - If the register is not valid, or the status byte is invalid.
+ * @return CC1120_ERROR_CODE_SUCCESS - If the read was successful.
+ * @return An error code - If the register is not valid, or the status byte is invalid.
  */
-bool cc1120_read_spi(uint8_t addr, uint8_t data[], uint8_t len) {
-    bool status = true;
+cc1120_error_code cc1120_read_spi(uint8_t addr, uint8_t data[], uint8_t len) {
+    cc1120_error_code status = CC1120_ERROR_CODE_SUCCESS;
     
     if (addr >= CC1120_REGS_EXT_ADDR) {
         mcu_log(CC1120_LOG_LEVEL_ERROR, "cc1120_read_spi: Not a valid register!\n");
-        status = false;
+        status = CC1120_ERROR_CODE_READ_SPI_INVALID_REGISTER;
```

Comment:
Errors like these can probably just be merged into a single type of error, like "CC1120_ERROR_CODE_INVALID_PARAM"